### PR TITLE
New field ClusterName in kubevirt providerconfig

### PR DESCRIPTION
**Pull request by dermorz**
**What this PR does / why we need it**:
This is needed for the machine-controller to set the new cluster labels. (see kubermatic/kubermatic#10923)

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
https://docs.kubermatic.com/kubermatic/main/architecture/supported-providers/kubevirt/kubevirt/
```
We have a draft document including all KubeVirt changes and a migration guide for KKP 2.22. That document will include the documentation for this feature.

### CODE REVIEW

**Comment by hdurand0710** [comment]
/hold 
Due to  pre-kubermatic-test-integration

**Review by mfranczy — APPROVED** [review]
/approve
/lgtm

**Comment by hdurand0710** [comment]
You might have to update now `CompleteKubevirtProviderSpec` to add the clusterName ?

**Review by hdurand0710 — CHANGES_REQUESTED** [review]
I suspect that the clusterName will not be in the MD on a create cluster flow.

Confirmed in the  pre-kubermatic-e2e-kubevirt-centos-1.25 test, the MD contains:
```
  spec:
      metadata:
        creationTimestamp: null
      providerSpec:
        value:
       //...
            auth:
              kubeconfig: ""
            clusterName: ""
```

File: pkg/machine/provider/kubevirt_test.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func TestCompleteKubevirtProviderSpec(t *testing.T) {
 				datacenter: &kubermaticv1.DatacenterSpecKubevirt{
 					DNSPolicy: "test-dnspolicy",
 				},
-				expected: cloneBuilder(goodMachine).WithDNSPolicy("test-dnspolicy"),
+				expected: cloneBuilder(goodMachine).WithDNSPolicy("test-dnspolicy").WithClusterName("testcluster"),
```
Comment (hdurand0710) [inline pkg/machine/provider/kubevirt_test.go]: s/"testcluster"/goodCluster.Name

File: pkg/machine/provider/kubevirt_test.go [hunk 2]
```diff
@@ -90,7 +90,7 @@ func TestCompleteKubevirtProviderSpec(t *testing.T) {
 					DNSPolicy: "test-dnspolicy",
 				},
 				inputSpec: cloneBuilder(defaultMachine).WithDNSPolicy("keep-me-kubevirt"),
-				expected:  cloneBuilder(goodMachine).WithDNSPolicy("keep-me-kubevirt"),
+				expected:  cloneBuilder(goodMachine).WithDNSPolicy("keep-me-kubevirt").WithClusterName("testcluster"),
```
Comment (hdurand0710) [inline pkg/machine/provider/kubevirt_test.go]: s/"testcluster"/goodCluster.Name

**Review by hdurand0710 — CHANGES_REQUESTED** [review]

**Review by hdurand0710 — APPROVED** [review]
/lgtm

**Pull request merged**